### PR TITLE
Populate example plugins page

### LINKED ISSUE #1: Extend section: high level design
High level design. Individual issues can be opened for each section. 

- [ ] Plugins
    - [x] Overview
    - [x] The Manifest (with reference / documentation parsed from source)
    - [x] Front-end
        - [x] Introduction - How front-end customization is done, broad overview.
        - [x] Quickstart - Hello world!
        - [x] API Reference - Which elements can be plugged into, etc.
        - [ ] Playground? - Seems like it would be feasible to create a playground environment for UI elements.
    - [x] Back-end
        - [x] Introduction - How back-end customization is done, broad overview.
        - [x] Quickstart - Hello world!
        - [x] API Reference - More or less [the godoc page](https://godoc.org/github.com/mattermost/mattermost-server/plugin), reskinned.
    - [ ] Examples
- [ ] Forking Mattermost

### PR DESCRIPTION
**Pull request by ccbrown**
I think this completes the first pass of the "extend" section and closes #1.

### CODE REVIEW

File: site/content/extend/plugins/example-plugins.md [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+
+To get started extending server-side functionality with plugins, take a look at our [server "Hello, world!" tutorial](../server/hello-world).
+
+## Web App "Hello, world!"
+
+To get started extending browser-side functionality with plugins, take a look at our [web app "Hello, world!" tutorial](../webapp/hello-world).
+
+## JIRA
+
+The [JIRA plugin for Mattermost](https://github.com/mattermost/mattermost-plugin-jira) creates a webhook that your JIRA server can use to post messages to Mattermost when issues are created:
```
Comment (ccbrown) [inline site/content/extend/plugins/example-plugins.md]: @jwilander might also be a good time to actually review this repo (https://github.com/mattermost/mattermost-plugin-jira). The code is all the same as it was before, but I added a bunch of tests, the Makefile, Travis-CI, and Codecov.

Comment (jwilander) [reply site/content/extend/plugins/example-plugins.md]: Already started looking at it as an example for how to structure the zoom plugin's backend 👍

File: site/content/extend/plugins/example-plugins.md [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
 subsection: Plugins
 weight: 40
 ---
 
-Construction
+# Example Plugins
+
+## Server "Hello, world!"
+
+To get started extending server-side functionality with plugins, take a look at our [server "Hello, world!" tutorial](../server/hello-world).
```
Comment (jasonblais) [inline site/content/extend/plugins/example-plugins.md]: Propose referring to it as a "Hello, world!" guide, for consistency with the server intro page.

(Or using tutorial in both places, 0/5 on the choice, 5/5 on consistency)

File: site/content/extend/plugins/example-plugins.md [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+
+## Web App "Hello, world!"
+
+To get started extending browser-side functionality with plugins, take a look at our [web app "Hello, world!" tutorial](../webapp/hello-world).
```
Comment (jasonblais) [inline site/content/extend/plugins/example-plugins.md]: Same here re: `tutorial` --> `guide`

**Comment by ccbrown** [comment]
@jasonblais Updated! Went with "tutorial".

**Pull request merged**